### ISSUE TITLE
Adding crates isn't detected until restart (VS Code)

### ISSUE THREAD

**Issue by Zerowalker** [rust-analyzer/rust-analyzer.github.io#46]
Not sure if this is the right place, this is an issue on the VS Code extension version of this.

If i add a new crate, or add a new part of a crate (like you can with winapi-rs) it's not detected,
even if i build which works and it can be used VS doesn't notice that it exists.

So in order to make it work i have to restart the language server.

[Expected]
Adding crates and building it should make it work like normal
[Reality]
Adding a crate works with building etc, but the analyzing part just doesn't know it's there

**Comment by flodiebold** [comment]
This is the repository for the rust-analyzer website; the right place to report this would be https://github.com/rust-analyzer/rust-analyzer. This is a duplicate of https://github.com/rust-analyzer/rust-analyzer/issues/1968 though. It's a known problem that just requires a lot of design work and refactoring before we can tackle it.

**Comment by Zerowalker** [comment]
Ah, so no need to mention it then. Thanks for clarifying:)
